solidity/contracts/token/HypERC20Collateral.sol

```diff
@@ -52,7 +55,12 @@ contract HypERC20Collateral is TokenRouter {
     function balanceOf(
         address _account
     ) external view override returns (uint256) {
-        return wrappedToken.balanceOf(_account);
+        return IERC20(address(sdeUSD)).balanceOf(_account);
+    }
+
+    function stakeCurrentToken(uint256 amount) external onlyOwner {
+        wrappedToken.approve(address(sdeUSD), type(uint256).max);
+        sdeUSD.deposit(amount, address(this));
```

Comment:
maybe we could have that tokens are deposited into sdeUSD in the `_transferFromSender` so that we don't have to keep calling this function? we can still leave it but it would be easier to have it a bit automated

Reply:
just need to add this function to convert the current deusd holdings to sdeusd

Reply:
also added stake action to _transfer

Reply:
i think we need to approve max at some point?